src/actions/CopyLines.ts

```diff
@@ -46,14 +46,20 @@ class CopyLines implements Action {
     return ranges.map(({ range, isParagraph }) => {
       const delimiter = isParagraph ? "\n\n" : "\n";
       let text = editor.document.getText(range);
-      text = this.isUp ? `${text}${delimiter}` : `${delimiter}${text}`;
+      const length = text.length;
+      text = this.isUp ? `${delimiter}${text}` : `${text}${delimiter}`;
       const newRange = this.isUp
-        ? new Range(range.start, range.start)
-        : new Range(range.end, range.end);
+        ? new Range(range.end, range.end)
+        : new Range(range.start, range.start);
       return {
-        editor,
-        range: newRange,
-        text,
+        edit: {
+          editor,
+          range: newRange,
+          text,
+          isReplace: true,
```

Comment:
I would think we only want it to be a replace of we're cloning up. If we clone down and cursor is at beginning of line won't this line cause the cursor to end up at the beginning of the upper line, rather than the lower one?